drivers/adc/adc_demo/adc_demo.c

```diff
@@ -85,6 +112,144 @@ int32_t adc_demo_remove(struct adc_demo_desc *desc)
 	return SUCCESS;
 }
 
+int32_t adc_get_number_of_channels(void* dev)
+* @brief close all channels
+* @param dev - physical instance of an adc device
+* @return SUCCESS in case of success.
+		  *******************************************************************************/
+		  int32_t close_adc_channels(void* dev)
+{
+	struct adc_demo_desc *desc;
+
+	if(!dev)
+		return -ENODEV;
+
+	desc=dev;
+
+	desc->active_ch = 0;
+
+	return SUCCESS;
+}
+
+/***************************************************************************//**
+ * @brief utility function for computing next upcoming channel
+ * @param ch_mask - active channels .
+ * @param last_idx -  previous index.
+ * @param new_idx - upcoming channel index, return param.
+ * @return 1 if there are more channels, 0 if done.
+*******************************************************************************/
+static bool get_next_ch_idx(uint32_t ch_mask, uint32_t last_idx,
+			    uint32_t *new_idx)
+{
+	last_idx++;
+	ch_mask >>= last_idx;
+	if (!ch_mask) {
+		*new_idx = -1;
+		return 0;
+	}
+	while (!(ch_mask & 1)) {
+		last_idx++;
+		ch_mask >>= 1;
+	}
+	*new_idx = last_idx;
+
+	return 1;
```

Comment:
should be consistent in returns (return 0 return 1  or return SUCCESS return -ENODEV) 

If there are no more channels you can just return -ENOENT or something like that. There are just a few functions that return something positive like snprintf which returns the length of the string. In this case i would go with typical return SUCCESS or return and error if there is no next ch index but this is not a strong request so i'll leave it up to you.

Reply:
Well, this may be for me, I first wrote this function in iio_demo: 
The reason to return 1 was just to be more human readable in:
while(get_next_ch_idx ...
instead of while (0 == get_next_ch_idx)

Reply:
any way, I think we can add this to util.h beacause the same logic may be used in new iio drivers